libtorch-ffi/src/Torch/Internal/GC.hs

```diff
@@ -46,6 +48,14 @@ dumpLibtorchObjects ::
   IO ()
 dumpLibtorchObjects age = c_showWeakPtrList (fromIntegral age)
 
+newtype HasktorchException = HasktorchException String
+  deriving (Show)
+
+instance Exception HasktorchException
+
+unsafeThrowableIO :: forall a m. MonadThrow m => IO a -> m a
+unsafeThrowableIO a = unsafePerformIO $ (pure <$> a) `catch` (\(CppStdException msg) -> pure . throwM $ HasktorchException msg)
```

Comment:
I like it!

Reply:
Maybe we should export `HasktorchException` somewhere in `Torch.GraduallyTyped`? So that end users wouldn't need to add libtorch-ffi as a dependency if they want to catch this exception.